velox/exec/fuzzer/MemoryArbitrationFuzzer.cpp

```diff
@@ -0,0 +1,674 @@
+  return PlanWithSplits{
+      plan,
+      {{probeScanId, fromConnectorSplits(probeSplits)},
+       {buildScanId, fromConnectorSplits(buildSplits)}}};
+}
+
+bool isTableScanSupported(const TypePtr& type) {
+  if (type->kind() == TypeKind::ROW && type->size() == 0) {
+    return false;
+  }
+  if (type->kind() == TypeKind::UNKNOWN) {
+    return false;
+  }
+
+  for (auto i = 0; i < type->size(); ++i) {
+    if (!isTableScanSupported(type->childAt(i))) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+void writeToFile(
+    const std::string& path,
+    const VectorPtr& vector,
+    memory::MemoryPool* pool) {
+  dwrf::WriterOptions options;
+  options.schema = vector->type();
+  options.memoryPool = pool;
+  auto writeFile = std::make_unique<LocalWriteFile>(path, true, false);
+  auto sink =
+      std::make_unique<dwio::common::WriteFileSink>(std::move(writeFile), path);
+  dwrf::Writer writer(std::move(sink), options);
+  writer.write(vector);
+  writer.close();
+}
+
+std::shared_ptr<connector::ConnectorSplit> MemoryArbitrationFuzzer::makeSplit(
+    const std::string& filePath) {
+  return std::make_shared<connector::hive::HiveConnectorSplit>(
+      kHiveConnectorId, filePath, dwio::common::FileFormat::DWRF);
+}
+
+std::vector<MemoryArbitrationFuzzer::PlanWithSplits>
+MemoryArbitrationFuzzer::hashJoinPlans(const std::string& tableDir) {
+  static const std::vector<core::JoinType> kJoinTypes = {
+      core::JoinType::kInner,
+      core::JoinType::kLeft,
+      core::JoinType::kFull,
+      core::JoinType::kLeftSemiFilter,
+      core::JoinType::kLeftSemiProject,
+      core::JoinType::kAnti};
+
+  const auto numKeys = randInt(1, 5);
+  const std::vector<TypePtr> keyTypes = generateKeyTypes(numKeys);
+  std::vector<std::string> probeKeys = makeNames("t", keyTypes.size());
+  std::vector<std::string> buildKeys = makeNames("u", keyTypes.size());
+  const auto probeInput = generateProbeInput(probeKeys, keyTypes);
+  const auto buildInput = generateBuildInput(probeInput, probeKeys, buildKeys);
+
+  std::vector<std::shared_ptr<connector::ConnectorSplit>> probeScanSplits;
+  for (auto i = 0; i < probeInput.size(); ++i) {
+    const std::string filePath = fmt::format("{}/probe{}", tableDir, i);
+    writeToFile(filePath, probeInput[i], writerPool_.get());
+    probeScanSplits.push_back(makeSplit(filePath));
+  }
+
+  std::vector<std::shared_ptr<connector::ConnectorSplit>> buildScanSplits;
+  for (auto i = 0; i < buildInput.size(); ++i) {
+    const std::string filePath = fmt::format("{}/build{}", tableDir, i);
+    writeToFile(filePath, buildInput[i], writerPool_.get());
+    buildScanSplits.push_back(makeSplit(filePath));
+  }
+
+  std::vector<PlanWithSplits> plans;
+  plans.reserve(kJoinTypes.size());
+  for (const auto& joinType : kJoinTypes) {
+    if (!isTableScanSupported(probeInput[0]->type()) ||
+        !isTableScanSupported(buildInput[0]->type())) {
+      continue;
+    }
+    plans.push_back(hashJoinPlan(
```

Comment:
It looks like we need value plan and table scan plan otherwise we might not run anything in some runs? Thanks!